Tests/SonarQube.MSBuild.Tasks.UnitTests/AttachBuildWrapperTests.cs

```diff
@@ -55,16 +61,70 @@ public void AttachBW_MissingFiles_Fails()
         }
 
         [TestMethod]
+        public void AttachBW_MissingConfigFile_TaskFails()
+        {
+            // Arrange
+            string binFolder = TestUtils.CreateTestSpecificFolder(this.TestContext);
+            string configFolder = TestUtils.CreateTestSpecificFolder(this.TestContext, "config");
+            DummyBuildEngine dummyEngine = new DummyBuildEngine();
+
+            TaskExecutionContext taskContext = new TaskExecutionContext(binFolder, configFolder, null /* output folder */, 0 /* returns success code */ );
+
+            // Remove the config file
+            File.Delete(taskContext.ConfigFilePath);
+
+            AttachBuildWrapper testSubject = CreateTestSubject(dummyEngine, binFolder, binFolder);
+
+            // Act
+            bool result = testSubject.Execute();
+
+            // Assert
+            Assert.IsFalse(result, "Expecting task to fail");
+
+            dummyEngine.AssertSingleMessageExists(SonarQube.MSBuild.Tasks.Resources.Shared_ConfigFileNotFound);
+            dummyEngine.AssertNoWarnings();
+
+            AssertLogFileDoesNotExist(taskContext);
+        }
+
+        [TestMethod]
+        public void AttachBW_MissingOutputDirectorySetting_TaskFails()
+        {
+            // Arrange
+            string binFolder = TestUtils.CreateTestSpecificFolder(this.TestContext);
+            string configFolder = TestUtils.CreateTestSpecificFolder(this.TestContext, "config");
+            string outputFolder = Path.Combine(binFolder, "output");
+            DummyBuildEngine dummyEngine = new DummyBuildEngine();
+
+            TaskExecutionContext taskContext = new TaskExecutionContext(binFolder, configFolder, null /* no output dir set */, 0 /* returns success code */ );
+
+            AttachBuildWrapper testSubject = CreateTestSubject(dummyEngine, binFolder, configFolder);
+
+            // Act
+            bool result = testSubject.Execute();
+
+            // Assert
+            Assert.IsFalse(result, "Expecting task to fail");
+            Assert.IsFalse(Directory.Exists(outputFolder), "Not expecting the output folder to have been created");
+
+            dummyEngine.AssertSingleErrorExists(SonarQube.MSBuild.Tasks.Resources.BuildWrapper_MissingOutputDirectory);
+            dummyEngine.AssertNoWarnings();
+
+            AssertLogFileDoesNotExist(taskContext);
+        }
+
+        [TestMethod]
         public void AttachBW_NotAttached_ExeReturnsSuccess_TaskSucceeds()
         {
             // Arrange
-            string rootBinFolder = TestUtils.CreateTestSpecificFolder(this.TestContext);
-            string outputFolder = Path.Combine(rootBinFolder, "output"); // expected location: does not exist
+            string binFolder = TestUtils.CreateTestSpecificFolder(this.TestContext);
```

Comment:
Pre-existing test: just fixed up to supply the extra paths that are now required. The same goes for the rest of the existing tests.